src/core/server/plugins/discovery/plugins_discovery.ts

```diff
@@ -115,10 +115,12 @@ function createPlugin$(path: string, log: Logger, coreContext: CoreContext) {
   return from(parseManifest(path, coreContext.env.packageInfo)).pipe(
     map(manifest => {
       log.debug(`Successfully discovered plugin "${manifest.id}" at "${path}"`);
+      const opaqueId = Symbol(manifest.id);
       return new PluginWrapper(
         path,
         manifest,
-        createPluginInitializerContext(coreContext, manifest)
+        opaqueId,
+        createPluginInitializerContext(coreContext, opaqueId, manifest)
```

Comment:
optional: does it make sense to switch to named arguments instead of positional ones when  `arguments.length >3` ? to simplify refactoring in future